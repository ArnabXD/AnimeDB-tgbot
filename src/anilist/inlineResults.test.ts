import { assertSnapshot } from "testing/snapshot.ts";
import getInlineResults from "./inlineResults.ts";

Deno.test("inline result (anime)", async (snap) => {
  const result = await getInlineResults("boku no pico");
  assertSnapshot(snap, result);
});

Deno.test("inline result (manga)", async (snap) => {
  const result = await getInlineResults("Tokyo Revengers", "MANGA");
  assertSnapshot(snap, result);
});
