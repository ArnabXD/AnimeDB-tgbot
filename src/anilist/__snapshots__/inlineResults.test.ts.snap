export const snapshot = {};

snapshot[`inline result (anime) 1`] = `
{
  Page: {
    media: [
      {
        averageScore: 34,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/anime/cover/small/1639.jpg",
        },
        description: "During summer vacation, Tamotsu is on break from work happens to visit a particular café. It is at this café that Tamotsu meets Pico, who works there for his grandfather. Tamotsu instantly falls for Pico, believing that Pico is a girl because of his feminine appearance. However, to his disbelief, Pico claims he actually is a guy. Undeterred, Tamotsu pursues a sexual relationship with Pico, to an extent where he even persuades Pico into wearing girl clothes. When Pico asks what is he to Tamotsu, he finds himself unable to answer...<br><br>\\r
[Written by MAL Rewrite]",
        format: "OVA",
        genres: [
          "Hentai",
        ],
        id: 1639,
        idMal: 1639,
        isAdult: true,
        status: "FINISHED",
        studios: {
          nodes: [
            {
              name: "Natural High",
            },
          ],
        },
        title: {
          english: null,
          romaji: "Boku no Pico",
        },
        trailer: null,
      },
      {
        averageScore: 36,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/anime/cover/small/5391.jpg",
        },
        description: "The crisp rays of summer sun find the effeminate Pico embroiled in yet another heated and lustful fling. While out biking, Pico meets and befriends Chico—a lively boy not much younger than him, but leagues behind in the affairs of the flesh. After the innocent Chico shows his sister in the midst of self-pleasure to Pico, Pico takes it upon himself to educate this confused youth.<br>
<br>
(Source: MAL Rewrite)",
        format: "OVA",
        genres: [
          "Hentai",
        ],
        id: 5391,
        idMal: 5391,
        isAdult: true,
        status: "FINISHED",
        studios: {
          nodes: [
            {
              name: "Natural High",
            },
          ],
        },
        title: {
          english: null,
          romaji: "Pico to Chico",
        },
        trailer: null,
      },
      {
        averageScore: 36,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/anime/cover/small/4866.jpg",
        },
        description: "The third installment of the Boku no Pico series, which takes sometime after the second OVA, Pico to Chico.<br><br>
Pico and Chico meet a mysterious young boy named CoCo, whom Pico at first believes is a girl. The three of them become friends, but soon CoCo believes he is becoming a strain on Pico and Chico's relationship and decides to distance himself from them. The three of them rekindle their friendship by the end of the OVA.",
        format: "OVA",
        genres: [
          "Hentai",
        ],
        id: 4866,
        idMal: 4866,
        isAdult: true,
        status: "FINISHED",
        studios: {
          nodes: [
            {
              name: "Natural High",
            },
          ],
        },
        title: {
          english: null,
          romaji: "Pico x CoCo x Chico",
        },
        trailer: null,
      },
      {
        averageScore: 40,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/anime/cover/small/6546.jpg",
        },
        description: "An edited version of the first OVA titled pico~ Boku no Chiisana Natsu Monogatari (pico: My Little Summer Story) was released on November 11, 2007. It is a re-edited version of the animation seen in the Boku no Pico OVA set to a new script, featuring content that is more appropriate for viewers under the age of 18.<br>
<br>
(Source: nekoshota.com)",
        format: "OVA",
        genres: [
        ],
        id: 6546,
        idMal: 6546,
        isAdult: false,
        status: "FINISHED",
        studios: {
          nodes: [
            {
              name: "Natural High",
            },
          ],
        },
        title: {
          english: null,
          romaji: "pico: Boku no Chiisana Natsu Monogatari",
        },
        trailer: null,
      },
    ],
    pageInfo: {
      currentPage: 1,
      hasNextPage: false,
      lastPage: 1,
      perPage: 25,
      total: 4,
    },
  },
}
`;

snapshot[`inline result (manga) 1`] = `
{
  Page: {
    media: [
      {
        averageScore: 81,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/manga/cover/small/bx102988-OoVJxQCH6fbR.jpg",
        },
        description: "Hanagaki Takemichi was once a low-life punk, but after biting off more than he could chew with a gang in middle school, he ran away from his old life. Now that he’s an adult, the gang that once terrorized him has grown into the largest gang in Tokyo, and they’ve started killing people–including his old girlfriend! After learning this news, a mysterious force on a train platform shoves Takemichi right in front of an approaching train. As his life flashes before his eyes, he blinks to find himself transported back to middle school, before he fled from the gang! Now he has the chance to right what went wrong all those years ago. Can he save his girlfriend’s life in the future–and what else might he change if he’s rewriting the past?
<br><br>
(Source: Seven Seas Entertainment)
<br><br>
<i>Notes:<br>
- Winner of the Shounen Manga category of the 44th annual Kodansha Manga Awards.<br>
- Includes 1 extra chapter.</i>",
        format: "MANGA",
        genres: [
          "Action",
          "Drama",
          "Romance",
          "Supernatural",
        ],
        id: 102988,
        idMal: 104565,
        isAdult: false,
        status: "RELEASING",
        studios: {
          nodes: [
          ],
        },
        title: {
          english: "Tokyo Revengers",
          romaji: "Tokyo卍Revengers",
        },
        trailer: null,
      },
      {
        averageScore: 54,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/manga/cover/small/bx114523-AmEswBWBw0dc.png",
        },
        description: null,
        format: "MANGA",
        genres: [
          "Action",
        ],
        id: 114523,
        idMal: null,
        isAdult: false,
        status: "FINISHED",
        studios: {
          nodes: [
          ],
        },
        title: {
          english: null,
          romaji: "Baki Gaiden: Revenge Tokyo",
        },
        trailer: null,
      },
      {
        averageScore: null,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/manga/cover/small/bx152895-1Pd9JGkJYRQ3.jpg",
        },
        description: \`A compilation focused on the youth of the Touman Members 
<br><br>
<u><b>Volume 1:</b> SO YOUNG</u><br>
"Zero"<br>
"In Those Days"<br>
"One for all"<br>
"Man-crush"<br>
"Twin to Dragon"<br>
"You're not my Type"\`,
        format: "MANGA",
        genres: [
          "Action",
          "Comedy",
          "Drama",
        ],
        id: 152895,
        idMal: null,
        isAdult: false,
        status: "RELEASING",
        studios: {
          nodes: [
          ],
        },
        title: {
          english: null,
          romaji: "Tokyo卍Revengers Full Color Tanpenshuu",
        },
        trailer: null,
      },
      {
        averageScore: 68,
        coverImage: {
          medium: "https://s4.anilist.co/file/anilistcdn/media/manga/cover/small/bx151997-4RNeculzkQPO.jpg",
        },
        description: null,
        format: "MANGA",
        genres: [
          "Action",
          "Drama",
        ],
        id: 151997,
        idMal: 149751,
        isAdult: false,
        status: "RELEASING",
        studios: {
          nodes: [
          ],
        },
        title: {
          english: null,
          romaji: "Tokyo卍Revengers: Baji Keisuke Kara no Tegami ",
        },
        trailer: null,
      },
    ],
    pageInfo: {
      currentPage: 1,
      hasNextPage: false,
      lastPage: 1,
      perPage: 25,
      total: 4,
    },
  },
}
`;
